src/clr-angular/forms/common/layout.ts

```diff
@@ -20,5 +21,9 @@ export class ClrLayout implements OnInit {
     if (this.layout && this.layoutService.isValid(this.layout)) {
       this.layoutService.layout = this.layout;
     }
+    // Only set the label width if it is a valid option
+    if (this.labelSize && !this.layoutService.isVertical() && this.layoutService.labelSizeIsValid(this.labelSize)) {
```

Comment:
This can be removed if you use the above getter/setter.

Reply:
Should it still assign it to the service only if the layout is non-vertical?

Reply:
This should just be removed. We set it on the service always, we just don't always need it for vertical.

Reply:
removed this one and added the validation on the setter. So in some cases labelSize can be set for vertical layouts?